tests/director/tests/formation_api_test.go

```diff
@@ -1519,6 +1519,168 @@ func TestFormationNotificationsWithApplicationOnlyParticipants(t *testing.T) {
+
+		t.Logf("Add webhook with type %q and mode: %q to application with ID %q", webhookType, webhookMode, app1.ID)
+		actualApplicationWebhook := fixtures.AddWebhookToApplication(t, ctx, certSecuredGraphQLClient, applicationWebhookInput, tnt, app1.ID)
+		defer fixtures.CleanupWebhook(t, ctx, certSecuredGraphQLClient, tnt, actualApplicationWebhook.ID)
+
+		formationName := "app-to-app-formation-name"
+		t.Logf("Creating formation with name: %q from template with name: %q", formationName, formationTmplName)
+		defer fixtures.DeleteFormationWithinTenant(t, ctx, certSecuredGraphQLClient, tnt, formationName)
+		formation := fixtures.CreateFormationFromTemplateWithinTenant(t, ctx, certSecuredGraphQLClient, tnt, formationName, &formationTmplName)
+
+		assertFormationAssignments(t, ctx, tnt, formation.ID, 0, nil)
+		assertFormationStatus(t, ctx, tnt, formation.ID, graphql.FormationStatus{Condition: graphql.FormationStatusConditionReady, Errors: nil})
+
+		t.Logf("Assign application 1 to formation %s", formationName)
+		defer fixtures.CleanupFormation(t, ctx, certSecuredGraphQLClient, graphql.FormationInput{Name: formationName}, app1.ID, graphql.FormationObjectTypeApplication, tnt)
+		assignReq := fixtures.FixAssignFormationRequest(app1.ID, string(graphql.FormationObjectTypeApplication), formationName)
+		var assignedFormation graphql.Formation
+		err = testctx.Tc.RunOperationWithCustomTenant(ctx, certSecuredGraphQLClient, tnt, assignReq, &assignedFormation)
+		require.NoError(t, err)
+		require.Equal(t, formationName, assignedFormation.Name)
+
+		expectedAssignments := map[string]map[string]fixtures.AssignmentState{
+			app1.ID: {app1.ID: fixtures.AssignmentState{State: "READY", Config: nil}},
+		}
+		assertFormationAssignments(t, ctx, tnt, formation.ID, 1, expectedAssignments)
+		assertFormationStatus(t, ctx, tnt, formation.ID, graphql.FormationStatus{Condition: graphql.FormationStatusConditionReady, Errors: nil})
+
+		t.Logf("Assign application 2 to formation %s", formationName)
+		defer fixtures.CleanupFormation(t, ctx, certSecuredGraphQLClient, graphql.FormationInput{Name: formationName}, app2.ID, graphql.FormationObjectTypeApplication, tnt)
+		assignReq = fixtures.FixAssignFormationRequest(app2.ID, string(graphql.FormationObjectTypeApplication), formationName)
+		err = testctx.Tc.RunOperationWithCustomTenant(ctx, certSecuredGraphQLClient, tnt, assignReq, &assignedFormation)
+		require.NoError(t, err)
+		require.Equal(t, formationName, assignedFormation.Name)
+
+		expectedAssignments = map[string]map[string]fixtures.AssignmentState{
+			app1.ID: {
+				app1.ID: fixtures.AssignmentState{State: "READY", Config: nil},
+				app2.ID: fixtures.AssignmentState{State: "READY", Config: nil},
+			},
+			app2.ID: {
+				app1.ID: fixtures.AssignmentState{State: "READY", Config: str.Ptr("{\"key\":\"value\",\"key2\":{\"key\":\"value2\"}}")},
```

Comment:
Why in this [commit](https://github.com/kyma-incubator/compass/pull/3017/commits/f9f1edddc80feaaa26a4090981d2a20d8057bf3c) in the `tests/director/tests/formation_api_test.go` file you changed the `config_pending` state to `ready`? Shouldn't that be `config_pending` and test the new functionality where the state is provided in the response body? In this test, we call `.../formation-callback/with-state/..` which should return `config_pending`? Am I missing something?

That's valid for the other assertions in this commit as well - lines 1578, 1627, and 1633

Reply:
Hmm, I think I found the reason for that - you changed the state in the ext-svc-mock in the Patch(and other handlers) from `config_pending` to `ready`? [source link](https://github.com/kyma-incubator/compass/pull/3017/commits/f9f1edddc80feaaa26a4090981d2a20d8057bf3c)
<img width="700" alt="image" src="https://user-images.githubusercontent.com/31803034/235716127-f114de68-ecb3-45ef-80e6-7bd20ee524e3.png">

Is there a specific reason for that? I think it's better to use a different state than `ready` because we already return "success status code" 200 and the "default behavior" for that is to set the assignment in a ready state? that way we cannot be sure from where the `ready` state comes from, is it from the state field in the response or either from the "old logic" that had?

Reply:
No, there is no reason. I switched it to READY because there were problems with the unassign method and I forgot to return it to READY.